doc/radosgw/dynamicresharding.rst

```diff
@@ -0,0 +1,79 @@
+===================================
+RGW Dynamic Bucket Index Resharding
+===================================
+
+.. versionadded:: Luminous
+
+A large bucket index can lead to performance problems. In order
+to address this problem we introduced bucket index sharding.
+Until Luminous changing the number of bucket shards (resharding)
+needed to be done offline, from Luminous we support
+online bucket resharding.
+
+Each bucket index shard can handle its entries efficiently up until
+reaching a certain threshold of entries. If this threshold is exceeded the system
+could encounters performance issues.
+The dynamic resharding feature detects this situation and increases
+automatically the number of shards used by the bucket index,
+resulting in the reduction of the number of entries in each bucket index shard.
+This process is transparent to the user.
+
+The detection process consists of two parts:
+The first part happens when new objects are added to the bucket,
+and the second is via a background process that periodically scans all the buckets.
+This is needed in order to deal with exisitng buckets in the system that are not being updated.
+A bucket that requires resharding is added to the reshard_log queue and will be
```

Comment:
Seems like 'reshard_log' could use backticks to be rendered in monospaced fonts, like \`\``reshard_log`\`\`